### PR TITLE
fix(MessagesList): Stop fetching when reaching the chat beginning

### PR DESCRIPTION
**Pull request by DorraJaouad**
### ☑️ Resolves

Old messages were fetched on every scroll event even though messages list weren't updated -> lost requests 

Create a flag to stop fetching when :
* We reach system messages conversation_created and history_cleared
* For chats that don't have such, we wait another request and compare if in both times the list didn't change then we confirm that the very beginning of the chat was reached ( this case applies to chats that have expiration messages, and history cleared after setting it)

### 🖼️ Screenshots / Screencasts
No visual changes just spinner doesn't show up when you scroll up at the top.

### CODE REVIEW

File: src/components/MessagesList/MessagesList.vue [hunk 1]
```diff
@@ -268,6 +272,7 @@ export default {
 		token(newToken, oldToken) {
 			// Expire older messages when navigating to another conversation
 			this.$store.dispatch('easeMessageList', { token: oldToken })
+			this.stopFetchingOldMessages = false
```
Comment (Antreesy) [inline src/components/MessagesList/MessagesList.vue:275]: Wanted to catch you here =)

File: src/components/MessagesList/MessagesList.vue [hunk 2]
```diff
@@ -689,7 +694,11 @@ export default {
...
 		 */
 		async getOldMessages(includeLastKnown) {
+			if (this.stopFetchingOldMessages) {
+				return
+			}
 			// Make the request
+			const prevMessagesList = this.messagesList
 			this.loadingOldMessages = true
 			try {
 				await this.$store.dispatch('fetchMessages', {
```
Comment (Antreesy) [inline src/components/MessagesList/MessagesList.vue:704]: we could also get a response here; then:
- check last message in the array `['conversation_created', 'history_cleared'].includes(response.data.ocs.data.at(-1)?.systemMessage)`
- check response status code (200 contains messages. 304 is empty - end of a list)

Comment (DorraJaouad) [reply src/components/MessagesList/MessagesList.vue:704]: Good point 🦅 , I assumed response code would be 200 for empty list too.

**Comment by nickvergessen** [comment]
> For chats that don't have such, we wait another request and compare if in both times the list didn't change then we confirm that the very beginning of the chat was reached ( this case applies to chats that have expiration messages, and history cleared after setting it)

This can also happen when you have a full page of "hidden system messages" e.g. reactions only

**Review by Antreesy — APPROVED** [review]
Tested for these 3 cases, should be enough to cover, and don't break anything

**Pull request merged**